packages/cli/src/cli/cogify/action.cog.ts

```diff
@@ -65,15 +66,29 @@ export class CommandCogCreate extends CommandLineAction {
     }
 
     const names = this.name?.values;
-    if (names == null) throw new LoggerFatalError({ names }, 'Names cannot be null');
-    for (const name of names) {
+    if (names != null) {
+      for (const name of names) {
+        if (!files.find((r) => r.name === name))
+          throw new LoggerFatalError(
+            { name, names: files.map((r) => r.name).join(', ') },
+            'Name does not exist inside job',
+          );
+        output.add(name);
+      }
+    }
+
+    const namesStr = this.names?.value;
+    if (namesStr == null) throw new LoggerFatalError({ names }, 'Names cannot be null');
+    const namesArr = namesStr.replace('[', '').replace(']', '').split(',');
```

Comment:
`JSON.parse`

should just use `--names` and have a `parseNames(s: string):string[]` which determines if its JSON like

eg `s.startsWith("[")` then `JSON.parse` ?